homeassistant/components/fritzbox_callmonitor/config_flow.py

```diff
@@ -0,0 +1,254 @@
+            return RESULT_INSUFFICIENT_PERMISSIONS
+        except FritzConnectionException:
+            return RESULT_INVALID_AUTH
+
+    async def _get_name_of_phonebook(self, phonebook_id):
+        """Return name of phonebook for given phonebook_id."""
+        phonebook_info = await self.hass.async_add_executor_job(
+            self._phonebook.fph.phonebook_info, phonebook_id
+        )
+        return phonebook_info[FRITZ_ATTR_NAME]
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return FritzBoxCallMonitorOptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, user_input=None):
+        """Handle configuration by yaml file."""
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=DATA_SCHEMA_USER, errors={}
+            )
+
+        self._host = user_input[CONF_HOST]
+        self._port = user_input[CONF_PORT]
+        self._password = user_input[CONF_PASSWORD]
+        self._username = user_input[CONF_USERNAME]
+
+        result = await self.hass.async_add_executor_job(self._try_connect)
+
+        if result == RESULT_INVALID_AUTH:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=DATA_SCHEMA_USER,
+                errors={"base": RESULT_INVALID_AUTH},
+            )
+
+        if result != RESULT_SUCCESS:
+            return self.async_abort(reason=result)
+
+        if result == RESULT_SUCCESS:
+            if CONF_PHONEBOOK in user_input:
+                self._phonebook_id = user_input[CONF_PHONEBOOK]
+                self._phonebook_name = user_input[CONF_NAME]
+
+            elif len(self._phonebook_ids) > 1:
+                return await self.async_step_phonebook()
+
+            else:
+                self._phonebook_id = DEFAULT_PHONEBOOK
+                self._phonebook_name = await self._get_name_of_phonebook(
+                    self._phonebook_id
+                )
+
+            await self.async_set_unique_id(
+                f"{self._serial_number}-{self._phonebook_id}"
+            )
+            self._abort_if_unique_id_configured()
+
+            return self._get_config_entry()
+
+    async def async_step_phonebook(self, user_input=None):
+        """Handle a flow to chose one of multiple available phonebooks."""
+        phonebooks = []
+
+        for phonebook_id in self._phonebook_ids:
+            phonebooks.append(await self._get_name_of_phonebook(phonebook_id))
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="phonebook",
+                data_schema=vol.Schema(
+                    {vol.Required(CONF_PHONEBOOK): vol.In(phonebooks)}
```

Comment:
Should we filter out the phonebooks that have already been set up?

Reply:
IMO it is more confusing for a user to filter out the already set up phonebooks. For example, you have two different phonebooks. You set up the first one via the drop down. However, while you set up the next one this drop down does not appear and it automatically sets up the second phonebook. 

So from a user perspective I would like to see consistency while setting up both my phonebooks e,g, display a drop down of all available phonebooks even if one or multiple of them have already been set up.

Reply:
One day we'll add support for disabled entries :)

Reply:
That sounds great! :)